numpy/core/src/multiarray/ctors.c

```diff
@@ -3726,7 +3726,9 @@ array_from_text(PyArray_Descr *dtype, npy_intp num, char *sep, size_t *nread,
     free(clean_sep);
 
 fail:
-    Py_DECREF(dtype);
+    if (r == NULL) {
+        Py_DECREF(dtype);
+    }
```

Comment:
this was the source of the other error report, and is a bit subtle: if `r` is an `ndarray` we do not need to decref

Reply:
Looks like `r === NULL` case is already handled up above. What am I missing?

Reply:
The DECREF can be done unconditioanlly here, I agree. I do not think that path can be taken at all.

Reply:
fixed